chain/chunks/src/lib.rs

```diff
@@ -1075,6 +1089,136 @@ impl ShardsManager {
         )
     }
 
+    fn prepare_partial_encoded_chunk_response_from_chunk(
+        &mut self,
+        request: PartialEncodedChunkRequestMsg,
+        chain_store: &mut ChainStore,
+        rs: &mut ReedSolomonWrapper,
+        chunk: ShardChunk,
+    ) -> Option<PartialEncodedChunkResponseMsg> {
+        // TODO(mina86): Is it correct to use rs.data_shard_count() and
+        // rs.total_shard_count() here?  Can those values change over time?
```

Comment:
Currently this is the number from 'num_blocks_producer_seats' in genesis.

We might want to change it in the future -- but that would be a larger change anyway (as we'd have to start storing this number somewhere in the header/epoch info)

Reply:
Do you have an idea for a better comment? Or should I just drop it altogether?

Reply:
I'd suggest using `self.runtime_adapter.num_total_parts()`, since that's how we access total number of parts throughout the code and we will reimplement this function if we decide to change `num_blocks_producer_seats` in the future.

Reply:
In the end `rs` is created from `self.runtime_adapter.num_total_parts()`:

```rust
impl Client {
    pub fn new(
        /* ... */
    ) -> Result<Self, Error> {
        /* ... */
        let data_parts = runtime_adapter.num_data_parts();
        let parity_parts = runtime_adapter.num_total_parts() - data_parts;
        /* ... */
        Ok(Self {
            /* ... */
            rs: ReedSolomonWrapper::new(data_parts, parity_parts),
            /* ... */
        })
    }
}
```

so sounds like using `rs` is correct here.  I’d rather not call
`self.runtime.num_total_parts()` and `num_data_parts()` here because
that would require creating a new `ReedSolomonWrapper` which, from
what I understand, is non-trivial operation.

My question and worry is whether number of parts can change over
time.  E.g. if someone requests a partial block from one year ago can
its encoding be different than encoding of a chunk created at head.

Reply:
Right now it doesn't since we never change the number of validator seats in our chain, but that is a very valid concern because that might and I think will change in the future. And that is why I suggested using `runtime_adapter.num_total_parts` here because if in the future we do change that, we will definitely need to change `num_total_parts`, so it's easy to track what needs to be changed in the code.

How about we use `num_total_parts` here but add an assert that it's the same value as `rs.total_shard_count`? For now, it's the same anyways. And this is more rust to future changes.